### PR TITLE
feat!: use npm for plugin operations

### LINKED ISSUE #642: [BUG] Yarn hoisting behaviour depends on plugin installation order
## Description

We are installing multiple plugins into the Salesforce CLI which uses this plugin.

Currently, it seems that changing the order we install plugins in changes yarn's hoisting behaviour, which can cause incorrect module versions to be required in a plugin.

## Expected behaviour

The order plugins are installed should not cause changes in execution behaviour.

## Current behaviour

The order plugins are installed causes dependencies to be incorrectly required, potentially causing significantly different behaviour depending on installation order.

## Screenshots
Failure case:
image

Success case: 
image

## Steps to reproduce

Success case Dockerfile: 
```
FROM node:18
RUN npm install -g @salesforce/cli 
RUN sf plugins install @salesforce/sfdx-scanner && \
    sf plugins install @salesforce/plugin-packaging && \
    sf plugins install @salesforce/plugin-community && \
    echo y | sf plugins install sfdx-git-delta && \
    sf plugins install sfdmu && \
    sf plugins install @salesforce/plugin-signups
```

Failure case Dockerfile:
```
FROM node:18
RUN npm install -g @salesforce/cli 
RUN sf plugins install @salesforce/sfdx-scanner && \
    sf plugins install @salesforce/plugin-packaging && \
    sf plugins install @salesforce/plugin-community && \
    sf plugins install @salesforce/plugin-signups && \
    sf plugins install sfdmu && \
    echo y | sf plugins install sfdx-git-delta
```

Build the dockerfiles and run any command from the @salesforce/sfdx-scanner plugin.

Command used when debugging this was `sfdx scanner:rule:add --language xml --path temp.xml`.

sfdx version information:
```
 CLI Version:
	@salesforce/cli/2.4.10

 Architecture:
	linux-x64

 Node Version:
	node-v18.17.1

 Plugin Version:
	@oclif/plugin-autocomplete 2.3.6 (core)
	@oclif/plugin-commands 2.2.21 (core)
	@oclif/plugin-help 5.2.16 (core)
	@oclif/plugin-not-found 2.3.36 (core)
	@oclif/plugin-plugins 3.2.0 (core)
	@oclif/plugin-search 0.0.22 (core)
	@oclif/plugin-update 3.1.30 (core)
	@oclif/plugin-version 1.3.8 (core)
	@oclif/plugin-warn-if-update-available 2.0.46 (core)
	@oclif/plugin-which 2.2.29 (core)
	@salesforce/cli 2.4.10 (core)
	apex 2.3.10 (core)
	auth 2.8.12 (core)
	community 2.3.11 (user)
	data 2.5.5 (core)
	deploy-retrieve 1.17.2 (core)
	info 2.6.38 (core)
	limits 2.3.29 (core)
	login 1.2.25 (core)
	org 2.9.30 (core)
	packaging 1.22.2 (user)
	schema 2.3.22 (core)
	settings 1.4.24 (core)
	signups 1.4.33 (user)
	sobject 0.2.2 (core)
	source 2.10.30 (core)
	telemetry 2.2.5 (core)
	templates 55.5.9 (core)
	trust 2.5.4 (core)
	user 2.3.28 (core)
	@salesforce/sfdx-scanner 3.15.0 (user)
	sfdmu 4.30.0 (user)
	sfdx-git-delta 5.24.2 (user)

 OS and Version:
	Linux 6.4.7-200.fc38.x86_64

 Shell:
	bash

 Root Path:
	/usr/local/lib/node_modules/@salesforce/cli
```

### LINKED ISSUE #710: can't use `NO_PROXY` env var if proxy is configured via `.yarnrc`
**Describe the bug**
can't use `NO_PROXY` env var when installing plugins if I set up proxy via yarn config.

`plugin-plugins` doesn't pass `--no-default-rc` flag to `yarn` when installing plugins so it will pick what's on `~/.yarnrc`. 
If I set the proxy address via `yarn config set`, then I can't use `NO_PROXY` due to a yarn v1 bug:
https://github.com/yarnpkg/yarn/issues/5048

**To Reproduce**
1. Set https proxy (don't need to setup a local proxy, can point to localhost): `yarn config set https-proxy "localhost:8888"`
2. try installing a plugin but set `NO_PROXY='registry.yarnpkg.com'` so that it can reach the yarn registry

**Expected behavior**
`NO_PROXY` is respected.

# Idea
`plugin-plugins` could pass `--use-yarnrc` to save a custom yarn config file that is savesd in the CLI data dir, that way users can still use the global yarn config file without being affected by CLI internals.
Also it should set `--no-default-rc` so that no global npmrc/yarnrc file is read. 

Also tried using `npm_config` env vars to pass proxy settings and passing `--no-default-rc` to yarn but it didn't work.

# Workarounds

1) only set proxy via env vars, no yarn/npm config.
2) follow this workaround: https://github.com/yarnpkg/yarn/issues/5048#issuecomment-604181595

**Screenshots**

### 1. no yarn/npm config vars, proxy env vars work fine:

Screenshot 2023-11-09 at 12 28 05

**NOTE**: see second run with the typo in `NO_PROXY` env var, that shows `HTTPS_PROXY` is respected.

### 2. proxy set in `.yarnrc` file, install of plugin-org fails when using `NO_PROXY` (can't reach registry)

Screenshot 2023-11-09 at 12 05 29

**NOTE**: in the screenshot I used the local sf CLI and removed plugin-trust to remove the preinstall hook (it does a request that fails b/c `localhost:8888` doesn't exist).

### 3. with suggested [workaround](https://github.com/yarnpkg/yarn/issues/5048#issuecomment-604181595)

Screenshot 2023-11-09 at 12 10 42

**Environment (please complete the following information):**
 - OS & version: macos sonoma
 - Shell/terminal & version zsh/alacritty

**Additional context**
Reported in Salesforce CLI repo:
https://github.com/forcedotcom/cli/issues/2540

### LINKED ISSUE #763: Provide more info about installation failure
## What

Provide more info about error that caused installation of plugin to fail

# What is the current problem

Recently I tried to install plugin in cli using `oclif` library (`salesforce/cli`). I have encountered error 

```
Installing plugin PLUGIN_NAME... failed
    Error: /home/myhome/.npm-packages/lib/node_modules/@salesforce/cli/node_modules/yarn/bin/yarn.js add PLUGIN_NAME --non-interactive 
    --mutex=file:/home/myHome/.local/share/sf/yarn.lock --preferred-cache-folder=/home/myHome/.cache/sf/yarn --check-files exited with code 1
```
This error does not provide any helpful text that would allow me to identify problem during installation.
Running command displayed by in above error resulted in:
```
/home/myHome/.npm-packages/lib/node_modules/@salesforce/cli/node_modules/yarn/bin/yarn.js add PLUGIN_NAME --non-interactive 
yarn add v1.22.21
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
error @salesforce/sf-plugins-core@5.0.13: The engine "node" is incompatible with this module. Expected version ">=18.0.0". Got "16.16.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

This error makes it clear to understand what went wrong and what needs to be fixed

### PR DESCRIPTION
**Pull request by mdonnalley**
### Breaking Changes
- use `npm` for all plugin operations.
  - Fixes #710
  - Fixes #642
  - Fixes https://github.com/oclif/plugin-plugins/issues/815
- `tsc` is no longer called on typescript ESM plugins installed from github. If you need that functionality, you should update your plugin's `prepare` script to build the project on install
- Removed export of `Plugins`
- Users relying on a `.yarnrc` configuration file will have to migrate to `.npmrc`

### Other Changes
- Renamed `oclif/plugins` debug scope to `@oclif/plugin-plugins`
- Rename `cli:yarn` debug scope to `@oclif/plugin-plugins:npm`
- `DEBUG` logs always contain verbose npm output, `--verbose` flag will send output to stdout. Fixes #763 

### Issues/Work Items
@W-14581559@ 
@W-14792145@

### CODE REVIEW

**Comment by mdonnalley** [comment]
## Self QA

```
~/.local/share/sf » sf -v
@salesforce/cli/2.25.7 darwin-arm64 node-v20.9.0
```

Install several plugins using old plugin-plugins (using all possible conventions)
```
sf plugins install \
  oclif/plugin-which \
  https://github.com/oclif/plugin-help.git \
  https://github.com/oclif/plugin-version.git#mdonnalley/shrinkwrap \
  org \
  @salesforce/plugin-release-management@dev \
  auth@3.3.3
```

Link this version of plugin-plugins
`sf plugins link ~/repos/oclif/plugin-plugins --no-install`

Test that users can update plugins and then install plugins with the new version (this is the most common scenario since `plugins update` runs after `sf update`)
✅ `sf plugins update`
✅ `sf plugins install dev`

Go back to old version
`sf plugins unlink @oclif/plugin-plugins`

Test that you can still do a successful install
✅ `sf plugins install dev`

Link this version of plugin-plugins
`sf plugins link ~/repos/oclif/plugin-plugins --no-install`

Test that you can still do a successful install
✅ `sf plugins install dev`

File: src/commands/plugins/install.ts [hunk 1]
```diff
@@ -135,39 +137,39 @@ e.g. If you have a core plugin that has a 'hello' command, installing a user-ins
...
         ux.action.stop(chalk.bold.red('failed'))
-        YarnMessagesCache.getInstance().flush(plugin)
         throw error
       }
 
       ux.action.stop(`installed v${plugin.version}`)
-
-      YarnMessagesCache.getInstance().flush(plugin)
-
-      this.log(chalk.green(`\nSuccessfully installed ${plugin.name} v${plugin.version}`))
```
Comment (iowillhoit) [inline src/commands/plugins/install.ts:170]: Did you mean to remove this log message?

Comment (mdonnalley) [reply src/commands/plugins/install.ts:170]: Yes. The spinner now shows the success (or failure) message

```
@salesforce/cli: Installing plugin @oclif/plugin-version@latest... installed v2.0.14
```

**Comment by iowillhoit** [comment]
## QA NOTES

> Note: The `qa` CLI has the `beta` version of `plugin-plugins`

### MISC 
- 🟢 Installs and updates are *significantly* faster and bit smaller with `npm`. This is largely due to not needed a "refresh" with yarn to lock dependencies. The more plugins you have installed, the larger these differences will be.
  - Using `@salesforce/plugin-source @salesforce/plugin-org @salesforce/plugin-user`, I am comparing `~/.local/share/sf/node_modules`
    - `yarn` 
      - First run took `41s`
      - Second run took `1m7s`
      - Size is `554Mb`
    - `npm` 
      - First run took `25s`
      - Second run took `19s`
      - Size is `411Mb`
  - @jshackell-sfdc this is worth calling out in RNs
- 🟢 When a `npm-shrinkwrap` is present, plugin dependencies seem to be much more predictable with `npm`. For example, running `sf plugins install @oclif/plugin-help @oclif/plugin-which @oclif/plugin-version` with the `qa` release of `sf`; you get nested `node_modules` directories for each plugin that contain the dependencies defined in their shrinkwrap file. This should prevent some of the odd hoisting issues we have seen in the past with `yarn add` and the `yarn install` refreshes.
- 🟢 Expanding on this ^, going from plugins installed with `yarn` to `npm` properly cleans up the top level `node_modules` directory and you get a nice set of nested deps :tada: 
- 🟢 Was able to create a prerelease of the `sf` CLI and all NUTs passed.
- 🔴 The `--silent` flag is now `true` by default. There is no way to get non-silent output.
  - `--silent=false` does not work on boolean flags. 
  - `allowNo` is not enabled
  - `--verbose` is not an equivalent of "non-silent"
- 🟢 The `--verbose` flag works.

### RELATED ISSUES
- 🔴 (yarn and npm) Running `sf plugins reset --reinstall` does not work with github url installs.
  - Run `sf plugins install https://github.com/oclif/plugin-version.git` 
    - Same behavior when using a branch or version (`#ew/testing-plugin-plugins-npm-branch-installs`)
  - Confirm version: `cat ~/.local/share/sf/package.json | jq .dependencies`
  - Run `sf plugins reset --reinstall`
  - Confirm version: `cat ~/.local/share/sf/package.json | jq .dependencies`
  - Notice that it is no longer a github url, but a npm package name
    - The reinstall message does show the correct url, despite installing from npm package: `Reinstalled @oclif/plugin-version (https://github.com/oclif/plugin-version.git)`
- 🟡 (yarn and npm) You can install from github url using a lot of different short-names, but you cannot use the same github shortnames to uninstall. For example: `sf plugins uninstall oclif/plugin-version` does not work. You have to uninstall with the npm package name (`@oclif/plugin-version`)
- 🟡 (yarn and npm) When a core plugin is manually installed (like `plugins-version`), running `sf plugins --core` does not show `(user)`, it just removes `(core)`. Would it be more clear if we added `(user)` when core plugins were "overwritten"
- 🟡 (yarn and npm) Unable to allow-list github url installs with `unsignedPluginAllowList.json`
- 🟡 (yarn and npm) When you install a specific version of a non-JIT plugin (like `@salesforce/plugin-org@3.4.5`, `sfdx-hardis@4.28.0`, or `https://github.com/oclif/plugin-version.git#2.0.11`), running `sf plugins update` it does not update the plugin to a newer version. This **does work** for JIT plugins. Maybe this is WAD because the user intentionally installed a different version of a non-JIT plugin, but at the very least a warning message would be nice. "Did not update plugin `@salesforce/plugin-org` because it was manually installed. 
  - For core plugins you could suggest: "Run `sf plugins uninstall @salesforce/plugin-org` to revert to bundled version."
  - For 3PP plugins you'd need to suggest installing it again without a branch or version: "Run `sfdx plugins install sfdx-hardis` to install the latest version."
  - Should we make core plugins work the same way as JIT? If the version of the core plugin is "behind", update it to match what is in the CLI's `package.json`. If it is ahead, show a warning. 
- 🔴 (yarn and npm) Odd uninstall behavior removes the wrong plugin, assuming this is a shortname bug
  - Run `sf plugins install @salesforce/plugin-org`
  - Run `sf plugins --core` 
    - Note: `org` does not have `(core)` next to it
  - Run `sf plugins uninstall @oclif/plugin-org`
    - Notice the `@oclif/`, not `@salesforce` ^
  - Run `sf plugins --core` 
    - Note: `@salesforce/plugin-org` was uninstalled. `org` has `(core)` next to it

### JIT PLUGINS
- 🟢 JIT plugins can be installed manually with `npm`
- 🟢 JIT plugins install when running the command for the first time with `npm`
- 🟢 JIT plugin installed with `yarn` can be updated with `npm`
- 🟢 Can reinstall a JIT plugin with `plugins reset --reinstall` with `npm`

### CORE NPM PACKAGE
- 🟢 Able to install `@salesforce/plugin-org` with npm
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `@salesforce/plugin-org` with yarn
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `@salesforce/plugin-org` with yarn
  - 🟢 Able to uninstall `@salesforce/plugin-org` with npm
  - 🟢 `sf plgugins --core` shows `org` as `(core)`
- 🟢 Able to install `@salesforce/plugin-org` with yarn
  - 🟢 Running `sf plugins reset --reinstall` reinstalls the correct version

### CORE NPM PACKAGE WITH VERSION
- 🟢 Able to install `@salesforce/plugin-org@3.4.5` with npm
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `@salesforce/plugin-org@3.4.5` with yarn
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `@salesforce/plugin-org@3.4.5` with yarn
  - 🟢 Able to uninstall `@salesforce/plugin-org@3.4.5` with npm
  - 🟢 `sf plgugins --core` shows `org` as `(core)`
- 🟢 Able to install `@salesforce/plugin-org@3.4.5` with yarn
  - 🟢 Running `sf plugins reset --reinstall` reinstalls the correct version

### 3PP PLUGIN NPM PACKAGE
- 🟢 Able to install `sfdx-hardis` with npm
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `sfdx-hardis` with yarn
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `sfdx-hardis` with yarn
  - 🟢 Able to uninstall `sfdx-hardis` with npm
- 🟢 Able to install `sfdx-hardis` with yarn
  - 🟢 Running `sf plugins reset --reinstall` works with npm 

### 3PP PLUGIN NPM PACKAGE WITH VERSION
- 🟢 Able to install `sfdx-hardis@4.28.0` with npm
  - - 🟡 `sf plugins update` with npm installs same version of plugin (see Related Issue above)
- 🟢 Able to install `sfdx-hardis@4.28.0` with yarn
  - - 🟡 `sf plugins update` with npm installs same version of plugin (see Related Issue above)
- 🟢 Able to install `sfdx-hardis@4.28.0` with yarn
  - 🟢 Able to uninstall `sfdx-hardis` with npm
- 🟢 Able to install `sfdx-hardis@4.28.0` with yarn
  - 🟢 Running `sf plugins reset --reinstall` installs the correct version with npm

### SHORTNAME PLUGINS
- 🔴 Shortname npm-package plugin installs not working on the first install. 
  - Delete all plugins 
  - With the npm version run `sf plugins install org`
  - You'll get `This plugin isn't signed by Salesforce. Only install the plugin if you trust its creator. Do you want to continue the installation?`. By adding a log to `plugin-trust` you can see that it is not correctly resolving `@salesforce/plugin-org`. It is just looking up `org` on npm.
    - 🟠 Related: We need to update the `plugin-trust` message to include the name of the plugin (instead of `This plugin`). This would also be nice when installing from github using a short name. For example if someone ran `sf plugins oclif/plugin-version` (forgot the `@`), it would be nice to know that this was installing from github and not `npm`. "Trying to install from https://github.com/oclif/plugin-version. [...]"
      - 🟡 You could also mention that signed plugin validation does not work on github urls: " Plugins installed from Github cannot be verified. Proceed? (y/N)"
    - ❓❓ What if we did away with shortnames in this major version? I have always found them convoluted. I think users should be explicit when installing code on their machines. 
  - It is worth noting that if you already have `org` installed via `yarn`, the `package.json` in the `~/.local/share/sf` dir will have `@salesforce/plugin-org` as a dependency. If you then try to install `org` using the `npm` version, it will correctly install and look up the signature for `@salesforce/plugin-org`
  - 🟡 Need to test shortname with a version (`org@3.4.0`)
- 🟢 Can uninstall using a shortname (`org`)
- 🟢 Able to install a shortname plugin with yarn
  - 🟢 Able to update a shortname plugin with npm
- 🟢 `sf plugins reset --reinstall` works with a shortname plugin 

### GITHUB ORG/REPO Installs
- 🟢 Able to install `oclif/plugin-version` with npm
  - 🟢 Able to update plugins (`oclif/plugin-version`) with npm
- 🟢 Able to install `oclif/plugin-version` with yarn
  - 🟢 Able to update plugins (`oclif/plugin-version`) with npm
- 🟢 Able to install `oclif/plugin-version` with yarn
  - 🟢 Able to uninstall `@oclif/plugin-version` with npm (note: this requires npm package name)
- 🟠 We should add a warning about the breaking change "`tsc` is no longer called on typescript ESM plugins installed from github." With something like `salesforcecli/plugin-org` (that does not have a compile as part of its `prepare` script) it appears to install correctly, but then none of the commands are recognized because the `lib` dir is not there.
- 🟢 Bonus fix! Github url installs with `yarn` could get confused and have duplicate `oclif` entries in the `~/.local/share/sf/package.json`. This does not appear to be an issue with `npm`.
  - Reproduce with yarn:
    - `sf plugins install oclif/plugin-version`
    - `sf plugins install https://githbu.com/oclif/plugin-version`
    - `cat ~/.local/share/sf/package.json`
  - 🟢 NOTE: If users are in this bad state with yarn installs (multiple entries), running a `sf plugins update` with `npm` fixes the `package.json`

### GITHUB FULL URL
- 🟢 Able to install `https://github.com/oclif/plugin-version` with npm
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version` with yarn
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version` with yarn
  - 🟢 Able to uninstall `@oclif/plugin-version` with npm (note: this requires npm package name)
- 🟡 `sf plugins reset --reinstall` does not work with github urls (see Related Issue above)

### GITHUB FULL URL (with .git)
- 🟢 Able to install `https://github.com/oclif/plugin-version.git` with npm
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version.git` with yarn
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version.git` with yarn
  - 🟢 Able to uninstall `@oclif/plugin-version` with npm (note: this requires npm package name)
- 🟡 `sf plugins reset --reinstall` does not work with github urls (see Related Issue above)

### GITHUB URL WITH BRANCH
- 🟢 Able to install `https://github.com/oclif/plugin-version#ew/testing-plugin-plugins-npm-branch-installs` with npm 
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version#ew/testing-plugin-plugins-npm-branch-installs` with yarn
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version#ew/testing-plugin-plugins-npm-branch-installs` with yarn
  - 🟢 Able to uninstall `@oclif/plugin-version` with npm (note: this requires npm package name)
- 🟡 Once the `plugin reset --reinstall` issue is fixed, confirm that it reinstalls the same branch

### GITHUB URL WITH BRANCH (with .git)
- 🟢 Able to install `https://github.com/oclif/plugin-version.git#ew/testing-plugin-plugins-npm-branch-installs` with npm 
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version.git#ew/testing-plugin-plugins-npm-branch-installs` with yarn
  - 🟢 Able to update plugin with npm
- 🟢 Able to install `https://github.com/oclif/plugin-version.git#ew/testing-plugin-plugins-npm-branch-installs` with yarn
  - 🟢 Able to uninstall `@oclif/plugin-version` with npm (note: this requires npm package name)
- 🟡 Once the `plugin reset --reinstall` issue is fixed, confirm that it reinstalls the same branch

### GITHUB URL WITH VERSION
- 🟢 Able to install `https://github.com/oclif/plugin-version.git#2.0.11` with npm 
  - 🟡 `sf plugins update` with npm installs same version of plugin (see Related Issue above)
- 🟢 Able to install `https://github.com/oclif/plugin-version.git#2.0.11` with yarn
  - 🟡 `sf plugins update` with npm installs same version of plugin (see Related Issue above)
- 🟢 Able to install `https://github.com/oclif/plugin-version.git#2.0.11` with yarn
  - 🟢 Able to uninstall `@oclif/plugin-version` with npm (note: this requires npm package name)
- 🟡 Once the `plugin reset --reinstall` issue is fixed, confirm that it reinstalls the same version

### LINKING PLUGINS
- 🟢 Able to link plugin with `.` with npm
- 🟢 Able to unlink with `sf plugins unlink` (in plugin dir)
  - 🟡 The output could maybe be improved here: `@salesforce/cli: Uninstalling .... done`. The first `.` is the current dir that you are unlinking
- 🟢 Able to link plugin with full path `~/dev/plugin-org` with npm
- 🟢 Able to uninstall plugin with shortname (`org`) with npm
- 🟢 Both `uninstall` and `unlink` remove a linked plugin with npm
- 🟢 Able to link a plugin with npm
  - 🟢 Able to update a linked plugin with npm
- 🟢 Able to link with yarn 
  - 🟢 Able to update a linked plugin with npm
- 🟢 `sf plugins reset --reinstall` works with a linked plugin

**Comment by mdonnalley** [comment]
> 🔴 The --silent flag is now true by default. There is no way to get non-silent output.

Fixed 

> 🔴 (yarn and npm) Running sf plugins reset --reinstall does not work with github url installs.

Fixed

> 🔴 (yarn and npm) Odd uninstall behavior removes the wrong plugin, assuming this is a shortname bug

Fixed 

> 🔴 Shortname npm-package plugin installs not working on the first install

Can't repro

> 🟠 We should add a warning about the breaking change "tsc is no longer called on typescript ESM...

Done.

```
❯ echo y | sf plugins install salesforcecli/plugin-org
 ›   Warning: @oclif/plugin-plugins is a linked ESM module and cannot be auto-transpiled. Existing compiled source will be used instead.
? This plugin isn't signed by Salesforce. Only install the plugin if you trust its creator. Do you want to continue the installation?, yes
Because you approved this plugin, you can avoid future installation confirmations by adding the plugin to the unsignedPluginAllowList.json file. For details, see https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_allowlist.htm.

up to date in 7s
 ›   Warning: This plugin from github may not work as expected because the prepare script did not produce all the expected files.
@salesforce/cli: Installing plugin https://github.com/salesforcecli/plugin-org... installed v3.5.0
```

**Comment by iowillhoit** [comment]
> 🔴 Shortname npm-package plugin installs not working on the first install
> 
> Can't repro

Huh, odd. This only seems to happens if the `~/.local/share/sf` dir does not exist... If the `sf` dir exists, even if it is empty, this works as expected. 

Screenshot 2024-03-22 at 2 41 02 PM

**Comment by iowillhoit** [comment]
## QA UPDATE

> 🔴 The --silent flag is now true by default. There is no way to get non-silent output.
> Fixed
- 🟢 Looks good! Non-silent output shows dep warnings. Using `--silent` works as expected

> 🔴 (yarn and npm) Running sf plugins reset --reinstall does not work with github url installs.
> Fixed
- 🟢 Confirmed. 
- 🟢 Works with and without the `--hard` flag
- 🟢 Can install via url with yarn, then `plugin reset --reinstall` with npm

> 🔴 (yarn and npm) Odd uninstall behavior removes the wrong plugin, assuming this is a shortname bug
> Fixed
- 🟢 No longer uninstalling the wrong plugin

> 🔴 Shortname npm-package plugin installs not working on the first install
> Can't repro
- :shrug: See comment above

> 🟠 We should add a warning about the breaking change "tsc is no longer called on typescript ESM...
> Done.
- 🟢 Confirmed, looks good.

**Comment by iowillhoit** [comment]
## QA UPDATE

> 🟡 Need to test shortname with a version (org@3.4.0)
- 🟢 Works as expected with npm

> 🟡 `sf plugins reset --reinstall` does not work with github urls (see Related Issue above)
- GITHUB FULL URL
  - 🟢 `sf plugins install https://github.com/oclif/plugin-version` with yarn
  - 🟢 `sf plugins reset --reinstall` with npm
- GITHUB FULL URL (with .git)
  - 🟢 `sf plugins install https://github.com/oclif/plugin-version.git` with yarn
  - 🟢 `sf plugins reset --reinstall` with npm
- GITHUB URL WITH BRANCH
  - 🟢 `sf plugins install https://github.com/oclif/plugin-version#ew/testing-plugin-plugins-npm-branch-installs` with yarn
  - 🟢 `sf plugins reset --reinstall` with npm
- GITHUB URL WITH BRANCH (with .git)
  - 🟢 `sf plugins install https://github.com/oclif/plugin-version.git#ew/testing-plugin-plugins-npm-branch-installs` with yarn
  - 🟢 `sf plugins reset --reinstall` with npm
- GITHUB URL WITH VERSION
  - 🟢 `sf plugins install https://github.com/oclif/plugin-version.git#2.0.11` with yarn
  - 🟢 `sf plugins reset --reinstall` with npm

**Comment by iowillhoit** [comment]
## QA UPDATE

- 🟢 Able to install with a custom registry 
  - Using verdaccio and `SF_NPM_REGISTRY=http://localhost:4873/`
- 🟢 Published a new version, it was installed with `sf plugins update`
- 🟢 Able to install with a custom registry with `yarn`
  - 🟢 Published a new version, it was installed with `sf plugins update` with `npm`
- 🟢 Disconnected from verdaccio and it fell back to npm's registry version of `plugin-org`

**Review by iowillhoit — APPROVED** [review]

**Pull request merged**